### ISSUE TITLE
calling another task twice

### ISSUE THREAD

**Issue by angelf** [erikhuda/thor#73]
Hi,

I need to create a task that runs another task N times. Invoke doesn't do the trick since the second task will be called only once. Is there any way to call thor tasks without using the invocation system?

 Thanks!

**Comment by antoniobeyah** [comment]
I have this same use case, can it be updated to validate the task name AND options?  I was bit by this in a similar fashion as @eventualbuddha.  I checked the code on master and it seems to be the same way.

(pseudocode)
method_option :yes, default: false, type: boolean
task migrate_user USER
def migrate_user(username)
  user = lookup_user(username)

  if !user.migrated?
    if options[:yes] || ask("Are you sure you want to migrate the user: #{username}?")
    migrate_user(username)
    invoke(:send_email, [username])
  else
    puts "user already migrated"
  end
end

task 'migrate_users'
method_option :yes, default: false, type: boolean
def migrate_users
  users = list_all_users
  puts users
  ask 'Are you sure you want to migrate the users listed above?'
  users.each { |user| invoke(:migrate_user, [user], yes: true }
end

I like the safety that thor gives of preventing migration of a user more than once and further of ensuring that if a user is migrated more than once they don't receive an email twice (from some other task for example). 

I can submit a PR if you feel this is a proper change.

**Comment by veloper** [comment]
For anyone coming here from google, this is how I solved it.

```
no_commands do 
  def execute(task, args, options)
    (klass, task) = Thor::Util.find_class_and_command_by_namespace(task)
    klass.new.invoke(task, args, options)
  end
end
```

use `execute` exactly like `invoke`

**Comment by Paul-Bob** [comment]
Is there any documentation about why can't `invoke` twice?

I found the line that's holding the execution https://github.com/rails/thor/blob/main/lib/thor/invocation.rb#L125 but can't understand why is designed like this.

I noticed that using `Rails::Generators.invoke`  instead just `invoke` works.

```ruby
def task
  Rails::Generators.invoke "avo:resource", ["Team"], {}   # Executed
  Rails::Generators.invoke "avo:resource", ["Course"], {} # Executed
end

def task
  invoke "avo:resource", ["Team"], {}   # Executed
  invoke "avo:resource", ["Course"], {} # Not Executed
end
```

Thank you
